libethereum/BlockChain.cpp

```diff
@@ -856,6 +856,8 @@ void BlockChain::rescue(OverlayDB& _db)
 				break;
 		}
 		catch (...) {}
+		if (l == 0)
```

Comment:
I'm not sure this is the right thing to do. The block with number 0 was deemed invalid above, so we should actually re-generate the genesis block.